src/wasm.rs

```diff
@@ -0,0 +1,130 @@
+use std::{
+    borrow::Cow,
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+use futures::{stream::SplitSink, Sink, Stream, StreamExt};
+use pharos::{Observable, ObserveConfig};
+use pin_project_lite::pin_project;
+use ws_stream_wasm::{WsErr, WsEvent, WsMessage, WsMeta, WsStream};
+
+use crate::websockets::WebsocketMessage;
+
+/// Creates a new pair of sink and stream which operate on [`WasmWebsocketMessage`] instead of `WsMessage` and `WsEvent` separately.
+pub async fn wasm_websocket_combined_split(
```

Comment:
Would we not prefer to wrap that in struct similar to the other clients we already have?

Reply:
Not sure what other clients you refer to?

`async-tungstenite` returns a `WebSocketStream` which we can just call `.split()` on to get the sink and stream we need for `AsyncWebsocketClientBuilder::build`, while `ws_stream_wasm` is a bit weird (it returns a `WsMeta` and a `WsStream`) so we need this trickery.